src/auth/negotiate/kerberos/negotiate_kerberos_pac.cc

```diff
@@ -362,6 +367,114 @@ getextrasids(char *ad_groups, uint32_t ExtraSids, uint32_t SidCount)
     return ad_groups;
 }
 
+static char *
+get_resource_group_domain_sid(uint32_t ResourceGroupDomainSid) {
+
+    if (ResourceGroupDomainSid!= 0) {
+        uint8_t rev;
+        uint64_t idauth;
+        char dli[256];
+        char *ag;
+        int l;
+
+        align(4);
+
+        uint32_t nauth = get4byt();
+
+        size_t length = 1+1+6+nauth*4;
+
+        ag=(char *)xcalloc((length+1)*sizeof(char),1);
+        // the first byte is a length of the SID
+        ag[0] = (char) length;
+        memcpy((void *)&ag[1],(const void*)&p[bpos],1);
+        memcpy((void *)&ag[2],(const void*)&p[bpos+1],1);
+        ag[2] = ag[2]+1;
+        memcpy((void *)&ag[3],(const void*)&p[bpos+2],6+nauth*4);
+
+        /* mainly for debug only */
+        rev = get1byt();
+        bpos = bpos + 1; /*nsub*/
+        idauth = get6byt_be();
+
+        snprintf(dli,sizeof(dli),"S-%d-%lu",rev,(long unsigned int)idauth);
+        for ( l=0; l<(int)nauth; l++ ) {
+            uint32_t sauth;
+            sauth = get4byt();
+            snprintf((char *)&dli[strlen(dli)],sizeof(dli)-strlen(dli),"-%u",sauth);
```

Comment:
Please handle `snprintf()` error return codes.

Reply:
I added assert() of the returned value in a new file version.